yaml/_dumper/dumper_state.ts

```diff
@@ -21,12 +21,12 @@ function compileStyleMap(
   const keys = Object.keys(map);
   let tag: string, style: StyleVariant;
   for (let index = 0, length = keys.length; index < length; index += 1) {
-    tag = keys[index];
+    tag = keys[index]!;
```

Comment:
Ditto. Can we tweak the for-loop condition to be `const tag of Object.keys(map)`?